### ISSUE TITLE
Printers fail to load when debugging

### ISSUE THREAD

**Issue by psteckler** [coq/coq#6566]
#### Version

The Coq Proof Assistant, version 8.8+alpha (January 2018)
compiled on Jan 8 2018 17:59:09 with OCaml 4.06.0

#### Operating system

Linux x64

#### Description of the problem

After starting `camldebug` in Emacs, the printers fail to load:
```
OCaml Debugger version 4.06.0

(ocd) source db
File /home/steck/.opam/4.06.0/lib/camlp5/gramlib.cma loaded
File /home/steck/.opam/4.06.0/lib/ocaml/threads/threads.cma loaded
File /home/steck/.opam/4.06.0/lib/ocaml/str.cma loaded
Cannot find file clib.cma
Error during code loading: error while linking /home/steck/coq-sources/master/coq/lib/lib.cma.
Reference to undefined global `Backtrace'
```

**Comment by psteckler** [comment]
With 8.7.1, there's a similar, but slightly different error:
```
OCaml Debugger version 4.06.0

(ocd) source db
File /home/steck/.opam/4.06.0/lib/camlp5/gramlib.cma loaded
File /home/steck/.opam/4.06.0/lib/ocaml/threads/threads.cma loaded
File /home/steck/.opam/4.06.0/lib/ocaml/str.cma loaded
File /home/steck/coq-sources/coq-8.7.1/coq/lib/clib.cma loaded
File /home/steck/coq-sources/coq-8.7.1/coq/lib/lib.cma loaded
Error during code loading: interface mismatch on Profile
(ocd) 
```

**Comment by ejgallego** [comment]
@psteckler , the issue on 8.7.1 is on your end [you didn't clean the tree enough for the rebuild].

However, for 8.8, the issue is legitimate and due to #6444 , where is the search path set for ocamldebug? Note that `clib.cma` now lives in `clib/`

**Comment by psteckler** [comment]
>  the issue on 8.7.1 is on your end 

I was starting with a clean untar of the source distribution. I first built with `-natdynlink no`, did a `make clean`, rebuilt without that option, same result.

I can try untarring again, if you think it might help.

**Comment by ejgallego** [comment]
Oh indeed 8.7.1 doesn't have the `Profile -> CProfile` renaming. That should be backported for 8.7.2 indeed.

**Comment by psteckler** [comment]
That renaming happened in #6038. Can that PR be back-ported to 8.7.2, or maybe just the rename can be cherry-picked?

**Comment by psteckler** [comment]
Pinging @Zimmi48 : maybe look at this back-porting.

**Comment by Zimmi48** [comment]
Thanks for the notification @psteckler. Unfortunately, I don't see how this commit can be backported given how this is a significant API change. I guess it just means that debugging with ocamldebug >= 4.05 and Coq < 8.8 will never be possible. What do other think? @herbelin @maximedenes

**Comment by ejgallego** [comment]
> Unfortunately, I don't see how this commit can be backported given how this is a significant API change.

Unfortunately indeed that seems tricky to backport correctly if we want plugins to work in 8.7.1 and 8.7.2, that is a pity as the problem is serious.

We should pack all Coq modules by the way, now it is done for plugins, and my jbuilder branch also does this automatically, maybe we can improve a bit the situation for `clib` at least.

**Comment by Zimmi48** [comment]
Fixed for 8.8 and won't fix for v8.7, unless someone can propose a good solution that doesn't break plugins.
